x/supply/client/rest/query.go

```diff
@@ -19,11 +28,40 @@ func RegisterRoutes(ctx context.CLIContext, r *mux.Router) {
 
 func circulatingSupplyHandler(ctx context.CLIContext) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		res, err := query.NewRawClient(ctx, types.ModuleName).CirculatingSupply()
-		if err != nil {
-			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
-			return
+		q := r.URL.Query().Get("q")
+		if len(q) == 0 {
+			res, err := query.NewRawClient(ctx, types.ModuleName).CirculatingSupply()
+			if err != nil {
+				rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
+				return
+			}
+			rest.PostProcessResponse(w, ctx, res)
+		} else {
+			res, err := query.NewClient(ctx, types.ModuleName).CirculatingSupply()
+			if err != nil {
+				rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
+				return
+			}
+			metric := sdk.Coins{}
+			if q == "total" {
+				metric = res.Total
+			} else if q == "circulating" {
+				metric = res.Circulating
+			} else {
+				rest.WriteErrorResponse(w, http.StatusBadRequest, "wrong q value")
+				return
+			}
+			supply := sdk.Coin{}
+			for _, coin := range metric {
+				if coin.Denom == denom {
```

Comment:
Can we use `StakingDenom` for the comparison here? This should be stitchable to any cosmos based network

Reply:
If not, let's take the denom in the query itself.